src/multibody/model.hpp

```diff
@@ -100,9 +100,13 @@ namespace se3
     std::vector<Frame> frames;
     
     /// \brief Vector of subtrees.
-    /// subtree[j] corresponds to the subtree supported by the joint j.
+    /// subtrees[j] corresponds to the subtree supported by the joint j.
```

Comment:
Both subtree and support are useless. Subtrees[j] is the range from j to lastChild[j]. Supports can be scanned following the parent map. 

You then can write equivalently
for( int isub=j; isub<=lastChild[j]; isub ++) ; // ... do something with the subtree
for ( int isup = j; isup>0; isup = parents[isup] ) ; // ... do something with the support.

Or
BOOST_FOREACH(int isub, subtrees[j]) ; // .. do something with the subtree
BOOST_FOREACH(int isup, support[j]) ; // .. do something with the support

To validate the use of these extra data, you need to time what is the extra cost of solution 1 wrt solution 2.